flaml/tune/analysis.py

```diff
@@ -120,9 +158,11 @@ def get_best_trial(
                 values are disregarded and these trials are never selected as
                 the best trial.
         """
+        if self.lexico_objectives is not None:
+            best_trial = self.lexico_best(self.trials)
+            return best_trial

```

Comment:
`self.lexico_objectives" is undefined.
This logic should be in the derived class, not here.